addon/components/async-button.js

```diff
@@ -4,8 +4,10 @@ import layout from '../templates/components/async-button';
 var get = Ember.get;
 var getWithDefault = Ember.getWithDefault;
 var set = Ember.set;
+var computed = Ember.computed;
 
 export default Ember.Component.extend({
+  hasBlock: computed.bool('template').readOnly(),
```

Comment:
Accessing `template` in 1.13 will cause a deprecation.  This polyfill does work on 1.12, but `positionalParams` [used here](https://github.com/dockyard/ember-cli-async-button/blob/master/addon/components/async-button.js#L9) will not work at all on versions prior to 1.13.

If this is useful even without `positionalParams` (maybe the params aren't needed in all cases?), then we should only add the polyfill when we are in Ember versions prior to 1.13.

If this is not useful without `positionalParams` then we shouldn't add the polyfill at all, and we should warn when using with an older Ember version.  Something like [this in ember-get-helpers](https://github.com/jmurphyau/ember-get-helper/blob/master/index.js#L18) or [this in ember-auto-register-helpers](https://github.com/stefanpenner/ember-cli-auto-register-helpers/blob/master/index.js#L18).